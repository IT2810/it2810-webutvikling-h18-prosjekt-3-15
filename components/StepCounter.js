import { Component } from 'react';
import { Pedometer } from "expo";
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default class StepCounter extends Component{

	constructor(props){
		super(props);

		this.state = {
			startDate: new Date(),
			endDate: new Date,
			stepCount: 0
		}
	}



	resetSteps(){
		this.setState({
			startDate: new Date()
		});
	}

	componentDidMount() {
		this.resetSteps();
		this._subscribe();
	}

	componentWillUnmount() {
		this._unsubscribe();
	}

	_subscribe = () => {
		this._subscription = Pedometer.watchStepCount(result => {
			// If resetStepCount is true, we'll set the current stepCount to what is current.
			Pedometer.getStepCountAsync(this.state.startDate, new Date()).then(
				result => {
					this.setState({
						stepCount: result.steps
					});
				}
			);


		});
	};
	_unsubscribe = () => {
		this._subscription && this._subscription.remove();
		this._subscription = null;
	};



	render() {
		return(
			<View>
				<Text>Steps: {this.state.stepCount}</Text>
			</View>
		)
	}

};