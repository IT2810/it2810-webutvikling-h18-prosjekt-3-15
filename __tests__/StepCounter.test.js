import React from 'react';
import StepCounter from '../components/StepCounter';
import renderer from 'react-test-renderer';


it('renders correctly', ()=>{
    const tree = renderer
        .create(<StepCounter/>)
        .toJSON();
    expect(tree).toMatchSnapshot()
});
it('check startState', ()=>{
    const tree = renderer.create(<StepCounter/>).getInstance();
    let x = tree.state.steps;
    expect(Object.is(x, 0));
});