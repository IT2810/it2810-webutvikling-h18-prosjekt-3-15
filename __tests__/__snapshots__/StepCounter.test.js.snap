// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#b0b4ba",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "padding": 26,
      }
    }
  >
    Steps taken today: 
    0
  </Text>
</View>
`;
